### PR TITLE
Blinky Demo for Cortex M3 MPS2 on Qemu

### PR DESCRIPTION
**Pull request by alfred2g**
Blinky Demo for Cortex M3 MPS2 on Qemu

Description
-----------
Implement the Blinky demo that runs on qemu for an emulated Cortex M3 MPS2 board

Test Steps
-----------
Ran Manually on Quemu:
```
parallels@parallels-Parallels-Virtual-Platform:~$ qemu-system-arm -machine mps2-an385  -monitor null -semihosting --semihosting-config enable=on,target=native -kernel /media/psf/Home/mps2_demo  -serial stdio -nographic
blinking
blinking
blinking
blinking
blinking
blinking
blinking
blinking
qemu-system-arm: terminating on signal 2
parallels@parallels-Parallels-Virtual-Platform:~$ 
```

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: FreeRTOS/Demo/CORTEX_M3_MPS2_QEMU_GCC/Readme.md [hunk 1]
```diff
@@ -0,0 +1,47 @@
+# Emulating MPS2 Cortex M3 AN385 on QEMU
+
+## Requirements
+1. GNU Arm Embedded Toolchain download [here](https://developer.arm.com/tools-and-software/open-source-software/developer-tools/gnu-toolchain/gnu-rm/downloads)
+3. qemu-arm-system downlaod [here](https://www.qemu.org/download)
```
Comment (cobusve) [inline FreeRTOS/Demo/CORTEX_M3_MPS2_QEMU_GCC/Readme.md:5]: Spelling - tsk tsk ...

File: FreeRTOS/Demo/CORTEX_M3_MPS2_QEMU_GCC/Readme.md [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+run: (make sure you build the debug version)
+```
+$ arm-none-eabi-gdb -q ./build/RTOSDemo.axf
+
+(gdb) target remote :1234
+(gdb) break main
+(gdb) c
+```
+## Demo
+This Demo implemets the blinky demo, the user should expect the word
```
Comment (cobusve) [inline FreeRTOS/Demo/CORTEX_M3_MPS2_QEMU_GCC/Readme.md:46]: Spelling

File: FreeRTOS/Demo/CORTEX_M3_MPS2_QEMU_GCC/Readme.md [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+```
+$ arm-none-eabi-gdb -q ./build/RTOSDemo.axf
+
+(gdb) target remote :1234
+(gdb) break main
+(gdb) c
+```
+## Demo
+This Demo implemets the blinky demo, the user should expect the word 
+"blinking" to be repeatedly prited on the screen
```
Comment (cobusve) [inline FreeRTOS/Demo/CORTEX_M3_MPS2_QEMU_GCC/Readme.md:47]: spelling

File: FreeRTOS/Demo/CORTEX_M3_MPS2_QEMU_GCC/main_blinky.c [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+#include <task.h>
+#include <queue.h>
+#include <stdio.h>
+
+static void prvQueueReceiveTask( void *pvParameters );
+static void prvQueueSendTask( void *pvParameters );
+extern int _write(int file, char *ptr, int len);
+
+#define mainQUEUE_RECEIVE_TASK_PRIORITY		( tskIDLE_PRIORITY + 2 )
+#define	mainQUEUE_SEND_TASK_PRIORITY		( tskIDLE_PRIORITY + 1 )
```
Comment (cobusve) [inline FreeRTOS/Demo/CORTEX_M3_MPS2_QEMU_GCC/main_blinky.c:38]: indentation

File: FreeRTOS/Demo/CORTEX_M3_MPS2_QEMU_GCC/main_blinky.c [hunk 5]
```diff
@@ -0,0 +1,131 @@
...
+		indefinitely provided INCLUDE_vTaskSuspend is set to 1 in
+		FreeRTOSConfig.h. */
+		xQueueReceive( xQueue, &ulReceivedValue, portMAX_DELAY );
+
+		/*  To get here something must have been received from the queue, but
+		is it the expected value?  If it is, toggle the LED. */
+		if( ulReceivedValue == ulExpectedValue )
+		{
+			_write(1, "blinking\n", 9);
+            vTaskDelay(1000);
```
Comment (cobusve) [inline FreeRTOS/Demo/CORTEX_M3_MPS2_QEMU_GCC/main_blinky.c:125]: indentation

**Review by cobusve — CHANGES_REQUESTED** [review]
I marked some spelling and style problems

**Review by cobusve — APPROVED** [review]

**Review by n9wxu — APPROVED** [review]

**Pull request merged**